src/miner.rs

```diff
@@ -100,10 +101,10 @@ impl MinerManager {
                 hashes_tried.fetch_add(1, Ordering::AcqRel);
 
                 if nonce.0 % 128 == 0 {
-                    match block_channel.try_recv() {
-                        Ok(new_state) => state = new_state,
-                        Err(TryRecvError::Empty) => (),
-                        Err(TryRecvError::Disconnected) => return Err(TryRecvError::Disconnected.into()),
+                    if (&block_channel).borrow().is_none()
+                        || ((&block_channel).borrow().as_ref().unwrap().id != state.as_ref().unwrap().id)
```

Comment:
hmm checking if `id` is not equal instead of bigger than mostly solves the overflow problem, interesting.

Reply:
also, can you please first borrow and then do the rest? the block_channel can be updated between the left side of the OR and the right side, and then it could panic on `unwrap`

Reply:
I now borrow before using (replaced three `(&block_channel).borrow()` with a single one)